tests/unit/local/lambda_service/test_local_lambda_invoke_service.py

```diff
@@ -61,6 +61,25 @@ def test_invoke_request_handler(self, request_mock, lambda_output_parser_mock, s
         lambda_runner_mock.invoke.assert_called_once_with('HelloWorld', '{}', stdout=ANY, stderr=None)
         service_response_mock.assert_called_once_with('hello world', {'Content-Type': 'application/json'}, 200)
 
+    @patch('samcli.local.lambda_service.local_lambda_invoke_service.LocalLambdaInvokeService.service_response')
+    @patch('samcli.local.lambda_service.local_lambda_invoke_service.LambdaOutputParser')
+    @patch('samcli.local.lambda_service.local_lambda_invoke_service.request')
+    def test_invoke_request_handler_async(self, request_mock, lambda_output_parser_mock, service_response_mock):
```

Comment:
We should be mocking out Thread here too and asserting the mock is called correctly.